storage/sql/template.go

```diff
@@ -97,7 +115,30 @@ func NewDataSourceBuilderFactoryFromTemplate(template SQLSourceTemplate) func([]
 	}
 }
 
+var offsetPrefix = []byte("sql_offset")
+
 func (ds *DataSource) Get(ctx context.Context, variables octosql.Variables, streamID *execution.StreamID) (execution.RecordStream, *execution.ExecutionOutput, error) {
+	tx := storage.GetStateTransactionFromContext(ctx).WithPrefix(streamID.AsPrefix())
+
+	rs := &RecordStream{
+		stateStorage: ds.stateStorage,
+		streamID:     streamID,
+		isDone:       false,
+		alias:        ds.alias,
+		batchSize:    ds.batchSize,
+	}
+	err := rs.loadOffset(tx)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "couldn't sql offset")
+	}
+
+	// Adding offset value to variables
+	offsetVariable := octosql.NewVariables(map[octosql.VariableName]octosql.Value{"offset": octosql.MakeInt(rs.offset)})
```

Comment:
change this to some magic name which the user is sure not to use maybe?

Overall I think we should change octosql.VariableName to provide a boolean if it's an internal variable, but this is definitely out of scope here :)

Reply:
Because offset is kinda normal